_overviews/core/collections-migration-213.md

```diff
@@ -0,0 +1,330 @@
+
+The new operations are provided via an implicit enrichment. You need to add the following import to make them
+available:
+
+~~~ scala
+import strawman.collection.decorators._
+~~~
+
+The following operations are provided:
+
+- `Seq`
+    - `intersperse`
+- `Map`
+    - `zipByKey` / `join` / `zipByKeyWith`
+    - `mergeByKey` / `fullOuterJoin` / `mergeByKeyWith` / `leftOuterJoin` / `rightOuterJoin`
+
+## Are there new implementations of existing collection types (changes in performance characteristics)?
+
+The default `Set` and `Map` are backed by a `ChampHashSet` and a `ChampHashMap`, respectively. The performance characteristics are the same but the
+operation implementations are faster. These data structures also have a lower memory footprint.
+
+`mutable.Queue` and `mutable.Stack` now use `mutable.ArrayDeque`. This data structure supports constant time index access, and amortized constant time
+insert and remove operations.
+
+## How do I cross-build my project against Scala 2.12 and Scala 2.13?
+
+Most usages of collections are compatible and can cross-compile 2.12 and 2.13 (at the cost of some warnings, sometimes).
+
+If you cannot get your code to cross-compile, there are various solutions:
+  - You can maintain a separate branch with the changes for 2.13 and publish releases for 2.13 from this branch.
+  - You can put source files that don't cross-compile in separate directories and configure sbt to assemble the sources according to the Scala version (see also the examples below):
+
+        // Adds a `src/main/scala-2.13+` source directory for Scala 2.13 and newer
+        // and a `src/main/scala-2.13-` source directory for Scala version older than 2.13
+        unmanagedSourceDirectories in Compile += {
+          val sourceDir = (sourceDirectory in Compile).value
+          CrossVersion.partialVersion(scalaVersion.value) match {
+            case Some((2, n)) if n >= 13 => sourceDir / "scala-2.13+"
+            case _                       => sourceDir / "scala-2.13-"
+          }
+        }
+
+  - You can use the [`scala-collection-compat`](https://github.com/scala/scala-collection-compat) library, which makes some of 2.13's APIs available to 2.11 and 2.12. This solution does not always work, for example if your library implements custom collection types.
+
+Note that the `scala-collection-compat` library has not fully stabilized yet. We expect that new, binary incompatible versions of this library will be published (for 2.11, 2.12) until Scala 2.13 is getting close to its final state. Therefore you might want to avoid adding a dependency on that library to your 2.11 / 2.12 artifacts for the time being.
+
+Examples of libraries that cross-compile with separate source directories:
+  - https://github.com/scala/scala-parser-combinators/pull/152
+  - https://github.com/scala/scala-xml/pull/222
+  - Some other examples are listed here: https://github.com/scala/community-builds/issues/710
+
+# Collection Implementers
```

Comment:
Should we just link to the other pages of the documentation?

https://docs.scala-lang.org/overviews/core/architecture-of-scala-213-collections.html
https://docs.scala-lang.org/overviews/core/custom-collections.html
https://docs.scala-lang.org/overviews/core/custom-collection-operations.html

Reply:
done too.